node/network/pov-distribution/src/lib.rs

```diff
@@ -121,45 +146,24 @@ async fn handle_signal(
 	state: &mut State,
 	ctx: &mut impl SubsystemContext<Message = PoVDistributionMessage>,
 	signal: OverseerSignal,
-) -> SubsystemResult<bool> {
+) -> error::Result<bool> {
 	match signal {
 		OverseerSignal::Conclude => Ok(true),
 		OverseerSignal::ActiveLeaves(ActiveLeavesUpdate { activated, deactivated }) => {
 			let _timer = state.metrics.time_handle_signal();
 
 			for relay_parent in activated {
-				let (vals_tx, vals_rx) = oneshot::channel();
-				ctx.send_message(AllMessages::RuntimeApi(RuntimeApiMessage::Request(
-					relay_parent,
-					RuntimeApiRequest::Validators(vals_tx),
-				))).await;
-
-				let n_validators = match vals_rx.await? {
-					Ok(v) => v.len(),
-					Err(e) => {
-						tracing::warn!(
-							target: LOG_TARGET,
-							err = ?e,
-							"Error fetching validators from runtime API for active leaf",
-						);
-
-						// Not adding bookkeeping here might make us behave funny, but we
-						// shouldn't take down the node on spurious runtime API errors.
-						//
-						// and this is "behave funny" as in be bad at our job, but not in any
-						// slashable or security-related way.
-						continue;
-					}
-				};
+				let n_validators = request_validators_ctx(relay_parent.clone(), ctx).await?.await??.len();
```

Comment:
we previously `continue`d on runtime error, what was the rationale for changing it to bail?

Reply:
The other call about `request_validators_ctx` should have been a `continue` not this one, i've fixed that.